plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -584,7 +585,9 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                   }
                   exhausted = block_is_exhausted();
                } else {
-                  _subjective_billing.subjective_bill_failure( first_auth, trace->elapsed, fc::time_point::now() );
+                   if (!trx->read_only)
```

Comment:
Actually, I think this is a bug in existing implementation. The `if` should be for `if ( ! disable_subjective_billing )`